src/clj/muuntaja/middleware.clj

```diff
@@ -37,19 +37,17 @@
 (defn wrap-params [handler]
   "Middleware that merges request `:body-params` into `:params`.
   Supports async-ring."
-  (fn
-    ([request]
-     (let [body-params (:body-params request)
-           request (if (map? body-params)
-                     (update request :params merge body-params)
-                     request)]
-       (handler request)))
-    ([request respond raise]
-     (let [body-params (:body-params request)
-           request (if (map? body-params)
-                     (update request :params merge body-params)
-                     request)]
-       (handler request respond raise)))))
+  (letfn [(set-params
+            ([{:keys [params body-params] :as request}]
```

Comment:
Clojure destucturing is great, but expensive. Changing this to:

```clj
([request]
  (let [body-params (:body-params request)
        params (:params request) ...
```

makes is ~30% faster with small inputs.